sharedbuffers/mapped_struct.py

```diff
@@ -1076,19 +1077,43 @@ def __len__(self):
         else:
             return self.bitlen
 
+    @cython.ccall
+    @cython.locals(dcode=cython.char, offset=cython.size_t)
+    def _contains(self, elem, dcode, offset):
+        xlen = len(self.objlist)
+        hint = xlen // 2
+        pindex = cython.cast(cython.p_char, self.objlist.pybuf.buf) + offset
+        if dcode == 'q':
+            return _c_search_hkey_i64(elem, pindex, 8, xlen, hint)
+        elif dcode == 'I':
+            return _c_search_hkey_ui32(elem, pindex, 4, xlen, hint)
+        elif dcode == 'i':
+            return _c_search_hkey_i32(elem, pindex, 4, xlen, hint)
+        elif dcode == 'd':
+            return _c_search_hkey_f64(elem, pindex, 8, xlen, hint)
+        else:
+            raise NotImplementedError("invalid frozenset data type %s" % chr(dcode))
```

Comment:
You do have to implement it. You could use bisect, or you could implement the relevant _c_search_hkey variants. It's mostly copypaste work.

Reply:
Oh, this will actually never be reached, as I checked before calling this method. I guess I should change the exception's message.